src/app/vm/vm-creation/template/vm-template-dialog.component.ts

```diff
@@ -40,11 +55,36 @@ export class VmTemplateDialogComponent extends TemplateFilterListComponent {
     this.preselectedTemplate = data.template;
   }
 
-  public onOk(template): void {
-    this.dialogRef.close(template);
+
+  public onOk() {
+    const data = Object.assign({} , this.selectedTemplate ? this.selectedTemplate : this.preselectedTemplate);
+
+    this.templateTagService.getAgreement(data)
+      .switchMap(res => {
+        if (res) {
+          return this.showTemplateAgreementDialog()
+        } else {
+          this.dialogRef.close(data);
```

Comment:
is it possible to leave only a single occasion of this.dialogRef.close(data) inside of subscribe?